tm2py/components/network/transit/transit_skim.py

```diff
@@ -507,6 +536,161 @@ def _calc_total_ivt(
+                num_processors=self.controller.num_processors,
+            )
+            self.controller.emme_manager.matrix_results(
+                spec2,
+                class_name=transit_class.name,
+                scenario=self.scenarios[time_period],
+                num_processors=self.controller.num_processors,
+            )
+            self.controller.emme_manager.matrix_results(
+                spec3,
+                class_name=transit_class.name,
+                scenario=self.scenarios[time_period],
+                num_processors=self.controller.num_processors,
+            )
+            self.controller.emme_manager.matrix_results(
+                spec4,
+                class_name=transit_class.name,
+                scenario=self.scenarios[time_period],
+                num_processors=self.controller.num_processors,
+            )
+
+            drive_perception_factor = self.config.drive_perception_factor
+            walk_speed = self.config.walk_speed
+            vot = self.config.value_of_time 
+            # divide drive time by mode specific perception factor to get the actual time
+            # for walk time, use walk distance/walk speed
+            # because the mode specific perception factors are hardcoded in the mode definition
+            spec_list = [
+                {
+                    "type": "MATRIX_CALCULATION",
+                    "constraint": None,
+                    "result": f'mf"{_tp_tclass}_DTIME"',
+                    "expression": f'mf"{_tp_tclass}_DTIME"/{drive_perception_factor}',
+                },
+                {
+                    "type": "MATRIX_CALCULATION",
+                    "constraint": None,
+                    "result": f'mf"{_tp_tclass}_DTIME"',
+                    "expression": f'mf"{_tp_tclass}_DTIME" - 60*mf"{_tp_tclass}_DTOLL"/{vot}',
+                },
+                {
+                    "type": "MATRIX_CALCULATION",
+                    "constraint": None,
+                    "result": f'mf"{_tp_tclass}_WAUX"',
+                    "expression": f'mf"{_tp_tclass}_WAUX"/({walk_speed}/60)',
+                },
+                {
+                    "type": "MATRIX_CALCULATION",
+                    "constraint": None,
+                    "result": f'mf"{_tp_tclass}_WACC"',
+                    "expression": f'mf"{_tp_tclass}_WACC"/({walk_speed}/60)',
+                },
+                {
+                    "type": "MATRIX_CALCULATION",
+                    "constraint": None,
+                    "result": f'mf"{_tp_tclass}_WEGR"',
+                    "expression": f'mf"{_tp_tclass}_WEGR"/({walk_speed}/60)',
+                },
+            ]
+            self.controller.emme_manager.matrix_calculator(
+                spec_list,
+                scenario=self.scenarios[time_period],
+                num_processors=self.controller.num_processors,
+            )
+
+    def skim_penalty_toll(
```

Comment:
Allow for skimming of a boarding time penalty, that can be set to zero for transfers.